app/controllers/profiles_controller.rb

```diff
@@ -141,15 +142,26 @@ def collections
       .joins('left join collections on collections.id = linked_id')
       .order(created_at: :desc)
 
-    @available_states = can?(:access_collections, @resource) ?
+    coauthored = Topics::EntryTopics::CollectionTopic
+      .where(linked_id: @resource.collection_roles.pluck(:collection_id))
+      .joins('left join collections on collections.id = linked_id')
+      .order(created_at: :desc)
```

Comment:
чтобы не дублировать код, тут и для `scope` выше удалить `joins` и `order`
```suggestion
    coauthored = Topics::EntryTopics::CollectionTopic
      .where(linked_id: @resource.collection_roles.pluck(:collection_id))
```

а ниже добавить удалённый код
```ruby
    @collection = QueryObjectBase.new(scope)
      .joins('left join collections on collections.id = linked_id')
      .order(created_at: :desc)
```